watertap/unit_models/electrodialysis_1D.py

```diff
@@ -778,6 +1070,497 @@ def eq_recovery_mass_H2O(self, t):
+            self.flowsheet().time,
+            self.diluate.length_domain,
+            self.ion_set,
+            doc="calcualte current density from the electrical input",
+        )
+        def eq_set_surface_conc(self, mem, side, t, x, j):
+            if not self.config.has_Nernst_diffusion_layer:
+                if (mem == "cem" and side == "cathode_left") or (
+                    mem == "aem" and side == "anode_right"
+                ):
+                    return (
+                        self.conc_mem_surf_mol_x[mem, side, t, x, j]
+                        == self.concentrate.properties[t, x].conc_mol_phase_comp[
+                            "Liq", j
+                        ]
+                    )
+                else:
+                    return (
+                        self.conc_mem_surf_mol_x[mem, side, t, x, j]
+                        == self.diluate.properties[t, x].conc_mol_phase_comp["Liq", j]
+                    )
+            else:
+                return Constraint.Skip
+
+        @self.Constraint(
+            self.membrane_set,
+            self.flowsheet().time,
+            self.diluate.length_domain,
+            doc="Calculate the total non-ohmic potential across an iem; this takes account of diffusion and Donnan Potentials",
+        )
+        def eq_potential_nonohm_membrane_x(self, mem, t, x):
+            if (
+                not self.config.has_Nernst_diffusion_layer
+            ) and x == self.diluate.length_domain.first():
+                self.potential_nonohm_membrane_x[mem, t, x].fix(0)
+                return Constraint.Skip
+
+            return self.potential_nonohm_membrane_x[mem, t, x] == (
+                Constants.gas_constant
+                * self.diluate.properties[t, x].temperature
+                / Constants.faraday_constant
+                * (
+                    sum(
+                        self.ion_trans_number_membrane[mem, j]
+                        / self.config.property_package.charge_comp[j]
+                        for j in self.cation_set
+                    )
+                    + sum(
+                        self.ion_trans_number_membrane[mem, j]
+                        / self.config.property_package.charge_comp[j]
+                        for j in self.anion_set
+                    )
+                )
+                * log(
+                    sum(
+                        self.conc_mem_surf_mol_x[mem, "cathode_left", t, x, j]
+                        for j in self.ion_set
+                    )
+                    / sum(
+                        self.conc_mem_surf_mol_x[mem, "anode_right", t, x, j]
+                        for j in self.ion_set
+                    )
+                )
+            )  # Note: the log argument is taken as *total ion concentration* while in the original equation as *electrolyte concentration*
+            # This is mathematically valid as the ratio eliminate the difference and electro-neutrality stands.
```

Comment:
```suggestion
            # This is mathematically valid as the ratio eliminate the difference and electro-neutrality stands.
```
Interesting - I _think_ I understand!